packages/ember-runtime/lib/mixins/container_proxy.js

```diff
@@ -55,7 +55,14 @@ export default Mixin.create({
    @param {Object} options
    @return {any}
    */
-  lookup: containerAlias('lookup'),
+  lookup() {
+    let container = get(this, '__container__');
+    let result = container.lookup(...arguments);
+    if (result && result.owner === undefined) {
+      result.owner = this;
```

Comment:
I agree. For better or worse we have already reserved `this.container`, and any top level "word" we take will potentially collide with something.

I'm not opposed to something better, but keeping it as `container` would preserve current `this.container.lookup` semantics and not take over another reserved word...